code/processes/gateway.q

```diff
@@ -193,7 +193,9 @@ deleteresult:{[queryid] .gw.results : (queryid,()) _ .gw.results}
 
 // add a result coming back from a server
 addserverresult:{[queryid;results]
- serverid:first exec serverid from .gw.servers where active, handle=.z.w;
+ serverid:$[@[get;`.finspace.enabled;0b];
+    first exec serverid from .gw.servers where handle=.z.w;
```

Comment:
does a server get removed from `.gw.servers` when it shuts down? if not we may have an issue when a new server comes in and is assigned same handle (I think this possibility of multiple servers in `.gw.servers` with same handle is probably why the `active` flag was included in this query to begin with)

Reply:
Hi @jonathonmcmurray ,
   You raise a good point. The function responsible for adding new servers to ```.gw.servers```, ```.gw.addserversfromconnectiontable``` is redefined in corresponding PR for TorQ-Finspace repo so that it only adds new servers with handles not already included in ```.gw.servers```.  Which means an inactive server with a live handle will not be readded into the table in Finspace. But this is only the case for if we are using the TorQ-Finspace app, hence the check if finspace is enabled

Reply:
I see. I don't think that scenario is very probable, but let me revisit it just in case

Reply:
> does a server get removed from `.gw.servers` when it shuts down? if not we may have an issue when a new server comes in and is assigned same handle (I think this possibility of multiple servers in `.gw.servers` with same handle is probably why the `active` flag was included in this query to begin with)

I'm sorry Jonathan, I went on a tangent without addressing the original question. Yes, on .z.pc, .gw.pc gets called, which basically sets the handle for that server to null - so if rdb3 get assigned handle 5, it should be added as an active server to ```.gw.servers``` as the handle associated with rdb1 is now set to null. Although in Finspace, it isn't always behaving as I expect due to what appears to me as some delays between cluster beginning it's shutdown and the kdb process actually disconnecting?